ra-optimisation/linear-range-action-rao/src/main/java/com/farao_community/farao/linear_range_action_rao/LinearRangeActionRao.java

```diff
@@ -52,68 +53,49 @@ public String getVersion() {
+    private PreContingencyResult createPreContingencyResultAndUpdateLinearRaoResult(Crac crac, LinearRangeActionRaoResult linearRangeActionRaoResult) {
+        List<MonitoredBranchResult> preContingencyMonitoredBranches = new ArrayList<>();
+        if (crac.getPreventiveState() != null) {
+            crac.getCnecs(crac.getPreventiveState()).forEach(cnec ->
+                preContingencyMonitoredBranches.add(createMonitoredBranchResultAndUpdateLinearRaoResult(cnec, linearRangeActionRaoResult)
+                ));
+        }
+        return new PreContingencyResult(preContingencyMonitoredBranches);
+    }
 
-                resultExtension.updateResult(margin); // update mininum margin and security status in LinearRangeActionRaoResult
+    private List<ContingencyResult> createContingencyResultsAndUpdateLinearRaoResult(Crac crac, LinearRangeActionRaoResult linearRangeActionRaoResult) {
+        List<ContingencyResult> contingencyResults = new ArrayList<>();
+        crac.getContingencies().forEach(contingency -> {
+            List<MonitoredBranchResult> contingencyMonitoredBranches = new ArrayList<>();
+            crac.getStates(contingency).forEach(state -> crac.getCnecs(state).forEach(cnec ->
+                contingencyMonitoredBranches.add(createMonitoredBranchResultAndUpdateLinearRaoResult(cnec, linearRangeActionRaoResult))));
+            contingencyResults.add(new ContingencyResult(contingency.getId(), contingency.getName(), contingencyMonitoredBranches));
+        });
+        return contingencyResults;
+    }
 
-                double maximumFlow = 0;
-                try {
-                    maximumFlow = cnec.getThreshold().getMaxThreshold().orElse(0.0);
-                } catch (SynchronizationException e) {
-                    LOGGER.error("Cannot get max threshold for cnec {}", cnec.getId());
-                }
-                double referenceFlow = maximumFlow - margin;
-                resultList.add(new MonitoredBranchResult(cnec.getId(), cnec.getName(), cnec.getCriticalNetworkElement().getId(), maximumFlow, referenceFlow, Double.NaN));
-            }
+    private MonitoredBranchResult createMonitoredBranchResultAndUpdateLinearRaoResult(Cnec cnec, LinearRangeActionRaoResult linearRangeActionRaoResult) {
+        double margin = systematicSensitivityAnalysisResult.getCnecMarginMap().getOrDefault(cnec, Double.NaN);
+        double maximumFlow = systematicSensitivityAnalysisResult.getCnecMaxThresholdMap().getOrDefault(cnec, Double.NaN);
+        if (Double.isNaN(margin) || Double.isNaN(maximumFlow)) {
+            throw new RuntimeException(format("Cnec %s is not present in the linear RAO result. Bad behaviour.", cnec.getId()));
         }
-        return resultList;
+        linearRangeActionRaoResult.updateResult(margin);
+        return new MonitoredBranchResult(cnec.getId(), cnec.getName(), cnec.getCriticalNetworkElement().getId(), maximumFlow, maximumFlow - margin, Double.NaN);
```

Comment:
could you also return in postOptimisationFlow `maximumFlow - margin` instead of `Double.NaN` ?

Reply:
even if there is no optimisation yet, the search-tree-rao reads the postOptimisationFlow to evaluate the cost of one search tree Leaf.

Reply:
ok !